### ISSUE TITLE
Authorization JWT bearer must not send "scope" in "assertion"

### ISSUE THREAD

**Issue by seliverstov-tinkoff** [golang/oauth2#461]
Our team have used [this jwt client](https://pkg.go.dev/golang.org/x/oauth2/jwt) for authorization by [RFC-7523](https://tools.ietf.org/html/rfc7523). In fact, the result "assertion" of this client has claim with a "scope", but there is no field "scope" in POST request like it described in [4.1  of RFC-7521](https://tools.ietf.org/html/rfc7521). 

I mean, request must look like this
```
POST /token.oauth2 HTTP/1.1
Host: authz.example.net
Content-Type: application/x-www-form-urlencoded

grant_type=urn%3Aietf%3Aparams%3Aoauth%3Agrant-type%3Ajwt-bearer
&scope=test
&assertion=eyJhbGciOiJFUzI1NiIsImtpZCI6IjE2In0.
eyJpc3Mi[...omitted for brevity...].
J9l-ZhwP[...omitted for brevity...]
```

**Comment by seliverstov-tinkoff** [comment]
i can make pull request if you are agree with me

**Comment by osttra-j-sernbrant** [comment]
Agreed, according to https://datatracker.ietf.org/doc/html/rfc7521#section-4.1 scope should be a form parameter not in the assertion.

**Comment by Wikivu** [comment]
bencede suan mantikli geliyo yetkilendirme

**Comment by mdarii** [comment]
hi @osttra-j-sernbrant I've created a PR to match the RFC, is possible to get a review for it?
